engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/FunctionBinding.scala

```diff
@@ -174,56 +175,137 @@ case object FunctionBinding extends IRPass {
               IR.Error.Conversion.MissingSourceType(args.head.name.name)
             )
           } else {
-            val firstArgumentType = args.head.ascribedType.get
-            val firstArgumentName = args.head.name
+            val firstArg :: restArgs = args
+            val firstArgumentType    = firstArg.ascribedType.get
+            val firstArgumentName    = firstArg.name
             val newFirstArgument =
               if (firstArgumentName.isInstanceOf[IR.Name.Blank]) {
                 val newName = IR.Name
                   .Literal(
-                    Constants.Names.THAT_ARGUMENT,
+                    if (restArgs.isEmpty) Constants.Names.THAT_ARGUMENT
+                    else Constants.Names.SELF_ARGUMENT,
```

Comment:
Can it happen that there is `IR.Name.Literal(SELF_ARGUMENT)`? Shouldn't that be `IR.Name.Self`?

Reply:
IrToTruffle handles both cases, but you are right that since this pass runs early in the pipeline it would make sense that we use `IR.Name.Self` instead. Surprisingly nothing blew up because of this